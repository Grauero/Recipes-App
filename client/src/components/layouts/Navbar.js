import React from 'react';
import PropTypes from 'prop-types';

import NavbarAuth from './NavbarAuth';
import NavbarUnAuth from './NavbarUnAuth';

const Navbar = ({ session }) => (
  <nav>
    {session && session.getCurrentUser ? <NavbarAuth session={session} /> : <NavbarUnAuth />}
  </nav>
);

Navbar.propTypes = {
  session: PropTypes.instanceOf(Object).isRequired
};

export default Navbar;
