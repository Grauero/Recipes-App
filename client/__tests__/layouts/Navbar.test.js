import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Navbar from '../../src/components/layouts/Navbar';

const props = { session: { getCurrentUser: { username: 'username' } } };
let component;

beforeEach(() => {
  component = shallow(<Navbar {...props} />);
});

afterEach(() => {
  component.unmount();
});

it('renders <NavbarAuth /> component when session.getCurrentUser is provided', () => {
  expect(toJSON(component)).toMatchSnapshot();
});

it('renders <NavbarUnAuth /> component when session.getCurrentUser is NOT provided', () => {
  component = shallow(<Navbar session={{}} />);

  expect(toJSON(component)).toMatchSnapshot();
});
