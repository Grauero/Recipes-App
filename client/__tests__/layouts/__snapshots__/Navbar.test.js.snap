// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <NavbarAuth /> component when session.getCurrentUser is provided 1`] = `
<nav>
  <NavbarAuth
    session={
      Object {
        "getCurrentUser": Object {
          "username": "username",
        },
      }
    }
  />
</nav>
`;

exports[`renders <NavbarUnAuth /> component when session.getCurrentUser is NOT provided 1`] = `
<nav>
  <NavbarUnAuth />
</nav>
`;
